domains/emissions/Emissions.API/Tests/CalculateSourceEmissionShareTest.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using API.Helpers;
+using API.Models;
+using API.Services;
+using Xunit;
+
+namespace Tests
+{
+    public sealed class CalculateSourceEmissionShareTest
+    {
+        readonly SourceEmissionShareDataSetFactory sourceEmissionShareDataSetFactory = new();
+
+        [Theory]
+        [InlineData(Aggregation.Total)]
+        [InlineData(Aggregation.Actual)]
+        [InlineData(Aggregation.Hour)]
+        [InlineData(Aggregation.Day)]
+        [InlineData(Aggregation.Month)]
+        [InlineData(Aggregation.Year)]
+        public void EmissionSharesAndMeasurements_CalculateTotalEmission_TotalAnRelativeEmission(Aggregation aggregation)
+        {
+            // Arrange
+            var dateFrom = new DateTime(2021, 1, 1, 22, 0, 0, DateTimeKind.Utc);
+            var dateTo = new DateTime(2021, 1, 2, 1, 59, 59, DateTimeKind.Utc);
+            var timeSeries = sourceEmissionShareDataSetFactory.CreateTimeSeries();
+            var emissionShares = sourceEmissionShareDataSetFactory.CreateEmissionsShares();
```

Comment:
```suggestion
            var timeSeries = datasetFactory .CreateTimeSeries();
            var emissionShares = datasetFactory.CreateEmissionsShares();
```

Reply:
We have two dataset factories. One for calculating the emissions and one for the source emissions.